### PR TITLE
Obtain the project_id from credentials if needed.

### PR DESCRIPTION
**Pull request by mr-salty**
If the project_id has not been set in the config file and cannot be
obtained via the metadata service, try to get it from the credentials.

Add a test for this, and change to using constants for the paths to the
credentials files.

### CODE REVIEW

File: lib/fluent/plugin/out_google_cloud.rb [hunk 1]
```diff
@@ -195,6 +195,14 @@ def configure(conf)
         fail Fluent::ConfigError, 'Unknown platform ' + @platform
       end
 
+      # If we still don't have a project ID, try to obtain it from the
+      # credentials.
+      if @project_id.nil?
+        @project_id = CredentialsInfo.project_id
```
Comment (filipjs) [inline lib/fluent/plugin/out_google_cloud.rb]: Where is "CredentialsInfo" coming from?

File: lib/fluent/plugin/out_google_cloud.rb [hunk 2]
```diff
@@ -564,6 +572,37 @@ def fetch_ec2_metadata
...
 
+    # TODO: This functionality should eventually be available in another
+    # library, but implement it ourselves for now.
+    module CredentialsInfo
+      # Determine the project ID from the credentials, if possible.
+      # Returns the project ID (as a string) on success, or nil on failure.
+      def self.project_id
+        return nil if @auth_method == 'private_key'
+        creds = Google::Auth.get_application_default(LOGGING_SCOPE)
+        if creds.issuer
```
Comment (filipjs) [inline lib/fluent/plugin/out_google_cloud.rb]: "issuer"
Is that expected or should be "isuser"?

**Comment by filipjs** [comment]
LGTM

File: lib/fluent/plugin/out_google_cloud.rb (hunk 2 above)
Comment (mr-salty) [inline lib/fluent/plugin/out_google_cloud.rb]: No, it's "issuer" - http://www.rubydoc.info/github/google/signet/Signet/OAuth2/Client#issuer-instance_method

**Pull request merged**